packages/yup/src/dateRange.js

```diff
@@ -0,0 +1,52 @@
+import moment from 'moment';
+
+export default function({ min, max, format = 'MM/DD/YYYY' }, msg) {
+  const minDate = moment(
+    min,
+    ['MM/DD/YYYY', format, 'MMDDYYYY', 'YYYYMMDD'],
+    true
+  );
+
+  const maxDate = moment(
+    max,
+    ['MM/DD/YYYY', format, 'MMDDYYYY', 'YYYYMMDD'],
+    true
+  );
+
+  // Can't use arrow function because we rely on 'this' referencing yup's internals
+  return this.test({
+    name: 'dateRange',
+    exclusive: true, // Validation errors don't stack
+    // NOTE: Intentional use of single quotes - yup will handle the string interpolation
+    message:
+      msg ||
+      `Date Range must be between ${minDate.format(
+        format
+      )} and ${maxDate.format(format)}`,
+    test(value) {
+      if (!value) return false;
+      let { startDate, endDate } = value;
```

Comment:
any way to make the object shape configurable (let the user define the start and end date keys)?

Reply:
Yeah we could like what we did for `reactstrap-validation-date` where we provide the key path to the start and end dates.